### PR TITLE
Refactor server integration tests

### PR DESCRIPTION
**Pull request by brfrn169**
Please take a look!

### CODE REVIEW

File: core/src/integration-test/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitAdminIntegrationTestBase.java [hunk 1]
```diff
@@ -42,12 +41,21 @@ public void afterAll() throws ExecutionException {
   protected final Properties getProperties() {
     Properties properties = new Properties();
     properties.putAll(getProps());
-    properties.setProperty(DatabaseConfig.TRANSACTION_MANAGER, "consensus-commit");
+    String transactionManager = properties.getProperty(DatabaseConfig.TRANSACTION_MANAGER, "");
+    if (!transactionManager.equals("grpc")) {
```
Comment (brfrn169) [inline core/src/integration-test/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitAdminIntegrationTestBase.java:45]: We need this for the integration tests for the `server` module.

File: server/src/integration-test/java/com/scalar/db/server/ConsensusCommitAdminIntegrationTestWithDistributedTransactionAdminService.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.Properties;
+import org.junit.jupiter.api.AfterAll;
+
+public class ConsensusCommitAdminIntegrationTestWithDistributedTransactionAdminService
+    extends ConsensusCommitAdminIntegrationTestBase {
+  private static final String PORT_FOR_SERVER_WITH_INCLUDE_METADATA_ENABLED =
+      Integer.toString(ServerConfig.DEFAULT_PORT + 1);
+
+  private ScalarDbServer server;
+  private ScalarDbServer serverWithIncludeMetadataEnabled;
```
Comment (brfrn169) [inline server/src/integration-test/java/com/scalar/db/server/ConsensusCommitAdminIntegrationTestWithDistributedTransactionAdminService.java:18]: This is for the tests for `include_metadata`.

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**